File: python/basics/while_loops.py
#!/usr/bin/env python3

# Demonstrating while loops in Python

# Simple while loop
print("Using a while loop:")
counter = 1
while counter <= 5:
    print(f"Counter: {counter}")
    counter += 1

# Nested while loops
print("\nUsing nested while loops:")
outer_counter = 1
while outer_counter <= 3:
    inner_counter = 1
    while inner_counter <= 3:
        print(f"Outer Counter: {outer_counter}, Inner Counter: {inner_counter}")
        inner_counter += 1
    outer_counter += 1

File: js/basics/while_loops.js
// filepath: js/basics/while_loops.js

// Demonstrating while loops in JavaScript
let count = 0;

console.log("While loop:");
while (count < 5) {
    console.log(`Count: ${count}`);
    count++;
}

console.log("While loop with break:");
count = 0;
while (count < 10) {
    if (count === 5) {
        console.log(`Breaking at ${count}`);
        break;
    }
    console.log(`Count: ${count}`);
    count++;
}

console.log("While loop with continue:");
count = 0;
while (count < 10) {
    if (count === 5) {
        console.log(`Skipping ${count}`);
        count++;
        continue;
    }
    console.log(`Count: ${count}`);
    count++;
}
